### ISSUE TITLE
[Solved ?] Exclude users/admins (based on their username) from seeing cached pages

### ISSUE THREAD

**Issue by trendymail** [sanderdekroon/WP-Rocket-Guest-Cache#7]
Hello!

Found a "solution" (your milage may vary) to exclude users/admins (based on their username) from seeing cached pages.

Long story short: when a user/admin is logged in, WordPress send/receive a cookie containing its username.

=> wordpress_logged_in_[hash]

So, it was quite simple to exclude theses users/admins from seeing cached pages.

I don't know (yet) how to send a Pull Request so the "raw" code is attached below (advanced-cache.php).

Any advice/hint would be greatly appreciated; :)

And, again, many thanks for this plugin!

[advanced-cache.php.txt](https://github.com/sanderdekroon/WP-Rocket-Guest-Cache/files/10097103/advanced-cache.php.txt)

**Comment by sanderdekroon** [comment]
I think your solution may work, but does require maintaining a list of administrators. I'd rater see we generate a list of administrators and export those to a file instead of manually maintaining a file. I think we can work that in later.

However, I did see some hardcoded paths regarding your development setup and the plugin header sees to be missing. Would you be willing to fix those issues and create an actual Pull Request? That way we can start integrating your solution so others may benefit from it :)

[Documentation on how to create a Pull Request can be found on the Github docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request). 

In short: fork this repository, commit your changes to your fork and then create a pull request from the Github interface.

**Comment by trendymail** [comment]
Hello!

> [...] does require maintaining a list of administrators [...]

Indeed but I need to exclude admins **and** some users.

If you only want to exclude administrators, please use [wp-cli](https://wp-cli.org).

Simple way get them:

`php wp-cli user list | grep administrator | tr "\t" " " | cut -d " " -f 2 | sort > admin_list.txt`

Many thanks for you anwser: I will create a Pull Request when I am sure everthing runs smoothly.

If it succeed, then plugin [No Cache for Admins](https://github.com/wp-media/wp-rocket-helpers/tree/master/cache/wp-rocket-no-cache-for-admins/) will no longer be needed (assuming all admins are excluded).

Have a great night. :)

**Comment by CyrilDesch** [comment]
> Hello!
> 
> Found a "solution" (your milage may vary) to exclude users/admins (based on their username) from seeing cached pages.
> 
> Long story short: when a user/admin is logged in, WordPress send/receive a cookie containing its username.
> 
> => wordpress_logged_in_[hash]
> 
> So, it was quite simple to exclude theses users/admins from seeing cached pages.
> 
> I don't know (yet) how to send a Pull Request so the "raw" code is attached below (advanced-cache.php).
> 
> Any advice/hint would be greatly appreciated; :)
> 
> And, again, many thanks for this plugin!
> 
> [advanced-cache.php.txt](https://github.com/sanderdekroon/WP-Rocket-Guest-Cache/files/10097103/advanced-cache.php.txt)

Work for me 👍 Thanks for the solution my friends

**Comment by trendymail** [comment]
You (@sanderdekroon): I don't have the time right now to implement such a feature. If you want, you are free to submit a PR.

Me (@trendymail): I don't have the time to tell you how I managed to put this feature in production. If you want, just ask.

Cheers!

PS: some answers were deleted by [sanderdekroon](https://github.com/sanderdekroon) so you might think I was angry at him.

I am not: one can delete "unwanted" message, this is life!

By the way, I still use this plugin (with some modifications) and it works great!

**Comment by meirkalmus** [comment]
any soltution for this?
i would like to prevent admins from being served by cache...

**Comment by sanderdekroon** [comment]
Unfortunately, no.
